src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoChildFactoryBase.cs

```diff
@@ -321,5 +322,10 @@ public override string GetNodeCustomName(object smoObject, SmoQueryContext smoCo
         {
             return string.Empty;
         }
+
+        public override Dictionary<string, object> GetExtraProperties(object smoObject, SmoQueryContext smoQueryContext)
+        {
+            return new Dictionary<string, object>();

```

Comment:
>new Dictionary<string, object>() [](start = 19, length = 32)

should this value be cached after first creation so (1) the same object is returned each time this method is called (2) we don't reallocate memory each time the function is called?

Reply:
..for the use case it looks like the properties are being added to the collection each time the derived method is called, so I guess you do need to reallocate each time with current design.

---
In reply to: [156536525](https://github.com/Microsoft/sqltoolsservice/pull/569#discussion_r156536525) [](ancestors = 156536525)